core/src/mender-artifact.c

```diff
@@ -696,6 +690,12 @@ artifact_read_manifest(mender_artifact_ctx_t *ctx) {
         const char *checksum_str = line;
         const char *filename     = separator + 2;
 
+        /* We do not support compressed artifacts */
+        if (mender_utils_strbeginswith(filename, "header.tar") && is_compressed(filename)) {
+            mender_log_error("Artifact compression is not supported");
+            return MENDER_FAIL;
+        }
```

Comment:
We need to check also for the payload file `data/0000.tar` (not the contents inside, but the payload which is what we treat it cannot be compressed neither)

Reply:
Can the payload file be compressed without the header being compressed? How can I test this, i.e. create an artifact with `-compression none` where the payload file is compressed?